contracts/Trustcoin.sol

```diff
@@ -29,7 +29,7 @@ contract Trustcoin is OutgoingMigrationTokenInterface, ERC20TokenInterface, Safe
   address public migrationMaster; // The Ethereum address which is allowed to set the new token's address
 
   mapping (address => uint256) public balances; // (ERC20)
-  mapping (address => mapping (address => uint256)) public allowed; // (ERC20)
+  mapping (address => mapping (address => int256)) public allowed; // (ERC20)
```

Comment:
also, we're converting int256 <=> uint256 a few times in the code. I think it's worth explaining in the comments why we believe it's safe (we're never going to overflow because the entire supply is too small, and we always make sure some `int256 x >= 0` before converting it to a `uint256`